src/Http/Http.Extensions/src/HttpRequestJsonExtensions.cs

```diff
@@ -0,0 +1,157 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics.CodeAnalysis;
+using System.IO;
+using System.Text;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+using Microsoft.Extensions.Primitives;
+
+#nullable enable
+
+namespace Microsoft.AspNetCore.Http.Json
+{
+    public static class HttpRequestJsonExtensions
+    {
+        [return: MaybeNull]
```

Comment:
This isn't the correct usage of `MaybeNull` because it applies to the `ValueTask<TValue>` not the `TValue`. I discussed this case with @terrajobst at one point at the conclusion was that there was no way to document the actual behavior.

Reply:
Good catch.